components/SegmentedControl/src/EqualPanel.cs

```diff
@@ -34,21 +34,33 @@ protected override Size MeasureOverride(Size availableSize)
     {
         maxItemWidth = 0;
         maxItemHeight = 0;
+        int UncollapsedChildrenCount = Children.Where(x => x.Visibility != Visibility.Collapsed).Count();
 
-        if (Children.Count > 0)
+        if (UncollapsedChildrenCount > 0)
         {
-            SetMaxDimensions(Children, availableSize);
+            foreach (var child in Children)
+            {
+                if (child.Visibility != Visibility.Collapsed)
```

Comment:
We're checking visibility on everything twice here, you can just grab the collection (remove the `.Count()`) of the uncollapsed children. Then in the if add the `.Count()` so that you can just iterate over the collection of uncollapsed children in your foreach loop here and remove the inner `if` visibility check.